### PR TITLE
#510 Implementation of adding regression and use case tests that include an app affiliation

### PR DESCRIPTION
**Pull request by codemascot**
#510 Implementation of adding regression and use case tests that include an app affiliation

## Description
#510 Implementation of adding regression and use case tests that include an app affiliation

## Related Issue(s)
- #510
- #507 

## Verification
- [x] **Your** code builds clean without any errors or warnings
- [ ] Manual testing done (required)
- [x] Relevant automated test added (if you find this hard, leave it and we'll help out)
- [x] All tests run green

### CODE REVIEW

**Comment by acn-sbuad** [comment]
isn't the same resource used in all environments? Would make more sense to have this as a repo secret rather than an env secret.

**Comment by acn-sbuad** [comment]
Is it a conscious decision to both org lookup and resource dependency to the existing test case rather than adding a new test with a seperate order request allowing us to validate the sending of an email to a known recipient isolated from sending to org and resource?

**Comment by acn-sbuad** [comment]
Would be good if we could document the org numbers in pedia along with the other test data.

**Comment by codemascot** [comment]
> isn't the same resource used in all environments? Would make more sense to have this as a repo secret rather than an env secret.

Yes, you are correct! 👍 This is more fit for being a repo secret than an environment variable. 

> Is it a conscious decision to both org lookup and resource dependency to the existing test case rather than adding a new test with a separate order request allowing us to validate the sending of an email to a known recipient isolated from sending to org and resource?

Yes! It was a  conscious decision! Well, although we thought about writing a different test at first, later we agreed to put it together with the existing one as this is a regression test. However, it is definitely worth a consideration again which approach would be better.

> Would be good if we could document the org numbers in pedia along with the other test data.

Already started working on it. Trying to find a better way to document it in AltinnPedia. 👍

**Comment by codemascot** [comment]
@acn-sbuad Here is the doc update: https://github.com/Altinn/altinnpedia/pull/148

File: test/k6/src/tests/orders_org_no.js [hunk 1]
```diff
@@ -0,0 +1,263 @@
...
+ * 04 - GET notification order with status
+ * 05 - GET email notification summary
+ * 06 - GET email notifications summary again after one minute for verification
+ */
+export default function (data) {
+  try {
+    if (data.runFullTestSet) {
+      var selfLink = TC01_PostEmailNotificationOrderRequest(data);
+      let id = selfLink.split("/").pop();
+      TC02_GetNotificationOrderById(data, selfLink, id);
```
Comment (acn-sbuad) [inline test/k6/src/tests/orders_org_no.js:239]: not sure I see the point of duplicating test caces 2-5 from the existing orders_email tests. Recipient being an organization doesn't change any of the logic related to retrieving an order.

Comment (SandGrainOne) [reply test/k6/src/tests/orders_org_no.js:239]: I also think that we can remove some of the duplicated tests.

Comment (codemascot) [reply test/k6/src/tests/orders_org_no.js:239]: I do agree! Will remove them today! 👍

Comment (codemascot) [reply test/k6/src/tests/orders_org_no.js:239]: I do agree that 02-04 we don't need! However we do need the 05 for the use case test, don't we? 🤔

**Review by SandGrainOne — APPROVED** [review]

**Pull request merged**